File: api/model/Titles.php
<?php
/**
 * Created by PhpStorm.
 * User: R_Rod
 * Date: 03/01/2019
 * Time: 17:58
 */

class Titles implements JsonSerializable
{
    private $title_id;
    private $name;
    private $rep_amount;
    private $color;

    /**
     * Titles constructor.
     * @param $title_id
     * @param $name
     * @param $rep_amount
     * @param $color
     */
    public function __construct($title_id, $name, $rep_amount, $color)
    {
        $this->title_id = $title_id;
        $this->name = $name;
        $this->rep_amount = $rep_amount;
        $this->color = $color;
    }

    /**
     * @return mixed
     */
    public function getTitleId()
    {
        return $this->title_id;
    }

    /**
     * @return mixed
     */
    public function getName()
    {
        return $this->name;
    }

    /**
     * @return mixed
     */
    public function getRepAmount()
    {
        return $this->rep_amount;
    }

    /**
     * @return mixed
     */
    public function getColor()
    {
        return $this->color;
    }


    /**
     * Specify data which should be serialized to JSON
     * @link https://php.net/manual/en/jsonserializable.jsonserialize.php
     * @return mixed data which can be serialized by <b>json_encode</b>,
     * which is a value of any type other than a resource.
     * @since 5.4.0
     */
    public function jsonSerialize()
    {
        return [
            "title_id" => $this->getTitleId(),
            "name" => $this->getName(),
            "rep_amount" => $this->getRepAmount(),
            "color" => $this->getColor()
        ];
    }
}

File: AppAndroid/HardwareHamlet/app/src/main/java/com/hardwarehamlet/hardwarehamlet/model/Titles.java
package com.hardwarehamlet.hardwarehamlet.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

@Entity
public class Titles {
    @PrimaryKey
    private int title_id;
    private String name;
    private int rep_amount;
    private String color;

    public Titles(int title_id, String name, int rep_amount, String color) {
        this.title_id = title_id;
        this.name = name;
        this.rep_amount = rep_amount;
        this.color = color;
    }

    public int getTitle_id() {
        return title_id;
    }

    public String getName() {
        return name;
    }

    public int getRep_amount() {
        return rep_amount;
    }

    public String getColor() {
        return color;
    }
}
